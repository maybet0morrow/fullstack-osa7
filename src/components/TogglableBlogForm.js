import React from "react"
import { Form, Button } from "semantic-ui-react"
import { notify } from "../reducers/notificationReducer"
import { createBlog } from "../reducers/blogReducer"
import { connect } from "react-redux"
import { generalButton } from "../styles"


class TogglableBlogForm extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            visible: false
        }
    }

    toggleVisibility = () => {
        this.setState({ visible: !this.state.visible })
    }

    handleSubmit = async (event) => {

        event.preventDefault()
        if(this.state.visible === false){
            event.target.title.value = ""
            event.target.author.value = ""
            event.target.url.value =""
            return
        }
        console.log("creating...")
        const title = event.target.title.value
        const author = event.target.author.value
        const url = event.target.url.value

        event.target.title.value = ""
        event.target.author.value = ""
        event.target.url.value =""

        if((title === "")||(author === "")||(url === "")){
            this.props.notify("Missing content",true,5)
            return
        }

        const blogObject = {
            title,
            author,
            url,
            likes: 0,
        }
        this.props.createBlog(blogObject)
        this.toggleVisibility()

    }




    render() {
        const hideWhenVisible = { display: this.state.visible ? "none" : "" }
        const showWhenVisible = { display: this.state.visible ? "" : "none" }
        return (
            <div>
                <div style={hideWhenVisible}>
                    <Button onClick={this.toggleVisibility} style={generalButton}> New Blog </Button>
                </div>
                <div style={showWhenVisible}>
                    <h2> Create a new Blog </h2>
                    <Form onSubmit={this.handleSubmit} >
                        <Form.Field >
                            <input name="title" placeholder="Title of the blog" />
                        </Form.Field>
                        <Form.Field>
                            <input name="author" placeholder="Authors name" />
                        </Form.Field>
                        <Form.Field>
                            <input name="url" placeholder="Link to the blog" />
                        </Form.Field>

                        <Button type="submit" style={generalButton}>Create</Button>
                        <Button type="cancel" onClick={this.toggleVisibility} style={generalButton}>Cancel</Button>

                    </Form>
                </div>
            </div>
        )
    }
}



export default connect(
    null,
    { notify, createBlog }
)(TogglableBlogForm)