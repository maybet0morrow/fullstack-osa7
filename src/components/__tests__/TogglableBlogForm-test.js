import React from "react"
import TogglableBlogForm from "../TogglableBlogForm"
import renderer from "react-test-renderer"
import thunk from "redux-thunk"
import configureStore from "redux-mock-store"
import { Adapter } from "../../setupTest"
import { shallow } from "enzyme"

//import store from "../../store"
const mockUser = {
    token: "asdasdasd",
    username: "blaa",
    name: "Jotaki"
}
const middlewares = [thunk]
const initialState = { user: mockUser }
const mockStore = configureStore(middlewares)


describe("SnapshotTest", () => {
    let store
    let wrapper
    beforeEach(() => {
        store = mockStore(initialState)
        wrapper = shallow(<TogglableBlogForm store={store} />)
    })
    it("renders correnctly", () => {
        const tree = renderer
            .create(wrapper)
            .toJSON()


        expect(tree).toMatchSnapshot()
    })
})