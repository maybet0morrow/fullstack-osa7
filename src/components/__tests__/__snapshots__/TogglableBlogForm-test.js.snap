// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapshotTest renders correnctly 1`] = `
<div>
  <div
    className="hideWhenVisible"
    style={
      Object {
        "display": "",
      }
    }
  >
    <button
      className="ui button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      style={
        Object {
          "backgroundColor": "#606060",
          "color": "white",
        }
      }
      tabIndex={undefined}
    >
       New Blog 
    </button>
  </div>
  <div
    className="showWhenVisible"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <h2>
       Create a new Blog 
    </h2>
    <form
      action={undefined}
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <input
          name="title"
          placeholder="Title of the blog"
        />
      </div>
      <div
        className="field"
      >
        <input
          name="author"
          placeholder="Authors name"
        />
      </div>
      <div
        className="field"
      >
        <input
          name="url"
          placeholder="Link to the blog"
        />
      </div>
      <button
        className="ui button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        style={
          Object {
            "backgroundColor": "#606060",
            "color": "white",
          }
        }
        tabIndex={undefined}
        type="submit"
      >
        Create
      </button>
      <button
        className="ui button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        style={
          Object {
            "backgroundColor": "#606060",
            "color": "white",
          }
        }
        tabIndex={undefined}
        type="cancel"
      >
        Cancel
      </button>
    </form>
  </div>
</div>
`;
